### PR TITLE
Momentum based gradient optimizer and optimization techniques for gradient descent

### LINKED ISSUE #4527: [Machine Learning] Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent
#### Proposed title of article
Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent

#### Proposed article introduction
Gradient Descent is an iterative optimization algorithm that finds the smallest value for a function. The general idea is to set the parameters to random values and then iterate in small steps in the direction of the `slope`. Gradient descent is widely used in supervised learning to minimize the error function and find the best parameter values.
Adaptive Optimization (A.O.) Algorithms are becoming more popular due to their ability to converge quickly. In contrast to the traditional Gradient Descent algorithm, all of these algorithms use statistics from previous iterations to strengthen the convergence process. We will be discussing it in detail in this article.
#### Key takeaways 
- Momentum-based Optimization
- Extensions  designed for gradient descent algorithm
- How does momentum help in gradient descent
- Optimization techniques
#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by NdoloBonface**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4527

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by NdoloBonface** [comment]
@lalith1403 
Here is the new PR
Duplicate of #4732

**Comment by hectorkambow** [comment]
@lalith1403 @WanjaMIKE 

Lets keep an eye on this PR - the author expressed interest in becoming a peer reviewer in the future. 

👍

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+date: 2021-11-19T00:00:00-18:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+ - url: /engineering-education/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/hero.jpg
+   alt: Gradient Descent Based Optimization Techniques example image
+---
+
+Alternative models can be built using Gradient Descent in machine learning systems. An objective function specifies the level of inaccuracy that a Machine Learning model will have on a given dataset during the training phase.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:18]: What do you mean by built using gradient descent? Gradient descent is used to minimize the loss function to learn the task appropriately.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+A random set of values will be used throughout training to maintain fairness. As a result, we get closer to the ideal function value with each repetition.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:20]: What values are you referring to?

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+- [Adaptive Moment Estimation](#adaptive-moment-estimation)
+- [RMSProp](#rmsprop)
+- [AdaGrad](#adagrad)
+- [AdaDelta](#adadelta)
+- [Mini Batch Stochastic Gradient Descent](#mini-batch-stochastic-gradient-descent)
+- [Various other optimizers that have emerged in the last few years](#various-other-optimizers-that-have-emerged-in-the-last-few-years)
+- [How to choose an optimizer](#how-to-choose-an-optimizer)
+
+### Optimization techniques
+For a brief while, we'll consider various optimization methods frequently employed in Deep Learning. In other words, if you came to this article hoping to learn the equation and mathematical explanation. We will also look at some Adaptive Optimization Algorithms (AOA). Compiling a program based on the current execution profile is known as adaptive optimization in computer science. You can easily make a tradeoff between JIT compilation and instruction interpretation in an adaptive optimizer solution.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:38]: Second sentence is incomplete.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+### How does momentum help in gradient descent
+This method works by detecting the direction of the steepest slope in its present state and then traveling in the direction of that sharpest slope to update its status.
+
+Momentum helps distinguish recent derivatives when Gradient Descent approaches the middle phase of the ravine, and so boosts the gradient descent's direction.
+
+Momentum is a prominent strategy used in conjunction with SGD. When searching, momentum collects the gradients of previous steps and the current ones to decide the best course of action. The following changes have been made to the gradient descent equations.
+###  Stochastic Gradient Descent 
+
+Using Stochastic Graduated Descent instead of Classic Gradient Descent, several of the flaws in the classic method are corrected By computing the derivative for each point on the trajectory it follows. It attempts to make up for its lack of computational rigor. It has more considerable noise than Gradient Descent. Hence it takes longer iterations to get to a minimal solution using SGD. In contrast to Gradient Descent, SGD computes the derivatives for all points at the same time.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:66]: Why is By capitalized?

**Review by lalith1403 — CHANGES_REQUESTED** [review]

**Comment by lalith1403** [comment]
There are some significant readability challenges in your article that raise concerns over how much review/editing would be required from our peer review team. We don't provide editing services and recommend that students leverage online editing services (there are both free and paid services that you can explore), or try engaging with the community or other peers in your network to have them review prior to submitting.  Thank you.

**Comment by hectorkambow** [comment]
>NOTE: A duplicate PR will go back into the queue and will not receive any further major revisions. A duplicate PR should be 100% ready to be published, otherwise it will be closed immediately. We encourage our new contributors to go over our [resources page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) and [new contributor folder](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) to be better familiar with the level of quality and style we are looking to publish.
 
Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
👍

**Comment by NdoloBonface** [comment]
@lalith1403 
All changes covered. Reopen so as to take effect. Thanks.

Cc @hectorkambow
